storages/python/python-storage-plugins/yaml.py

```diff
@@ -74,12 +73,17 @@ def load(self, id: str):
         Returns:
             A DLite Instance corresponding to the given `id` (UUID).
         """
-        return instance_from_dict(
+        inst = instance_from_dict(
             self._data,
             id,
             single=self.options.single,
             check_storages=False,
         )
+        # Metadata in single-entity form is always read-only
+        if inst.is_meta and self.single:
+            self.writable = False
+
+        return inst
```

Comment:
Is this ensuring the instance cannot be overwritten? This confuses me a bit. Here we've already loaded the instance, right? So setting the `writable` property afterwards - does that change anything?

Reply:
Yes, it makes a difference. It is not the instance we want to write-protect, but the yaml file. 

It is first after we have loaded the instance we know whether it is a metadata in single-entity form.